src/components/playback/playqueuemanager.js

```diff
@@ -127,6 +155,12 @@ define([], function () {
         var currentPlaylistItemId = this.getCurrentPlaylistItemId();
         var isCurrentIndex = playlistItemIds.indexOf(currentPlaylistItemId) !== -1;
 
+        if (this._sortedPlaylist.length <= playlistItemIds.length) {
+            this._sortedPlaylist = this._sortedPlaylist.splice(0).filter(function (item) {
+                return playlistItemIds.indexOf(item.PlaylistItemId) === -1;
+            });
+        }
```

Comment:
`playlistItemIds` is items to remove, so `this._sortedPlaylist.length <= playlistItemIds.length` doesn't work.
Also I see no reason to use `splice`.
```suggestion
        this._sortedPlaylist = this._sortedPlaylist.filter(function (item) {
            return !playlistItemIds.includes(item.PlaylistItemId);
        });
```

Reply:
@dmitrylyzo The condition is wrong, correct. I added it after having it working and, although I retested everything before pushing, for some reason it seems that I missed this.

* ``splice`` is required, otherwise the resulting array will be empty. In this function, the playlist variable (returned from ``getCurrentPlaylist`` is already spliced, that's why not used later. Just tested and it doesn't work without splice.

* Also, we must check against -1 index, no clue why but your return doesn't work as well.

What I did here is replicate with the sorted playlist array what it's done in the current playlist. It's not the ideal solution but we're too late in the feature freeze and this is not yet migrated to ES6, so I think it makes sense to leave all the refactoring for later (if nobody comes first, I want to migrate the playback components to ES6, already got some appreciation to them :P) and stay in the safe side, even in the little details. Didn't inspect the objects deeply as well to improve this, will do on ES6 migration.

Reply:
>     * `splice` is required, otherwise the resulting array will be empty.

Hmm, but it works without it. Moreover, `filter` returns a filtered copy.

Reply:
I see what is the problem now 👍